### PR TITLE
feat(StatusCheckbox): introduced new `leftSide` prop

### PR DESCRIPTION
**Pull request by saledjenic**
`leftSideCheckbox` determines whether to display check box on the left or right side, defaulted to left

### CODE REVIEW

File: src/StatusQ/Controls/StatusCheckBox.qml [hunk 1]
```diff
@@ -8,16 +8,20 @@ import StatusQ.Components 0.1
 
 
 CheckBox {
-    id: statusCheckBox
+    id: root
+
+    property bool leftSideCheckbox: true
```
Comment (noeliaSD) [inline src/StatusQ/Controls/StatusCheckBox.qml:13]: I suggest to change the name as it is indeed a property of the `CheckBox` component. It seems redundant for me. WDYT?

```suggestion
    property bool leftSide: true
```

File: src/StatusQ/Controls/StatusCheckBox.qml [hunk 2]
```diff
@@ -8,16 +8,20 @@ import StatusQ.Components 0.1
...
 
     indicator: Rectangle {
+        anchors.left: root.leftSideCheckbox? parent.left : undefined
```
Comment (noeliaSD) [inline src/StatusQ/Controls/StatusCheckBox.qml:16]: ;)
```suggestion
        anchors.left: root.leftSideCheckbox ? parent.left : undefined
```

File: src/StatusQ/Controls/StatusCheckBox.qml [hunk 3]
```diff
@@ -8,16 +8,20 @@ import StatusQ.Components 0.1
...
+        anchors.right: !root.leftSideCheckbox? parent.right : undefined
```
Comment (noeliaSD) [inline src/StatusQ/Controls/StatusCheckBox.qml:17]:
```suggestion
        anchors.right: !root.leftSideCheckbox ? parent.right : undefined
```

File: src/StatusQ/Controls/StatusCheckBox.qml [hunk 4]
```diff
@@ -8,16 +8,20 @@ import StatusQ.Components 0.1
...
         implicitWidth: 18
         implicitHeight: 18
-        x: statusCheckBox.leftPadding
+        x: !root.leftSideCheckbox? root.rightPadding : root.leftPadding
```
Comment (noeliaSD) [inline src/StatusQ/Controls/StatusCheckBox.qml:20]:
```suggestion
        x: !root.leftSideCheckbox ? root.rightPadding : root.leftPadding
```

File: src/StatusQ/Controls/StatusCheckBox.qml [hunk 5]
```diff
@@ -26,19 +30,21 @@ CheckBox {
...
+        text: root.text
+        font.pixelSize: root.font.pixelSize
         opacity: enabled ? 1.0 : 0.3
         verticalAlignment: Text.AlignVCenter
         wrapMode: Text.WordWrap
         width: parent.width
         color: Theme.palette.directColor1
-        leftPadding: !!statusCheckBox.text ? statusCheckBox.indicator.width + statusCheckBox.spacing
-                                           : statusCheckBox.indicator.width
+        leftPadding: root.leftSideCheckbox? (!!root.text ? root.indicator.width + root.spacing
```
Comment (noeliaSD) [inline src/StatusQ/Controls/StatusCheckBox.qml:45]:
```suggestion
        leftPadding: root.leftSideCheckbox ? (!!root.text ? root.indicator.width + root.spacing
```

File: src/StatusQ/Controls/StatusCheckBox.qml [hunk 6]
```diff
@@ -26,19 +30,21 @@ CheckBox {
...
+                                 : root.indicator.width) : 0
+        rightPadding: !root.leftSideCheckbox? (!!root.text ? root.indicator.width + root.spacing
```
Comment (noeliaSD) [inline src/StatusQ/Controls/StatusCheckBox.qml:47]:
```suggestion
        rightPadding: !root.leftSideCheckbox ? (!!root.text ? root.indicator.width + root.spacing
```

**Review by noeliaSD — APPROVED** [review]
LGTM. Just added some minor comments and also could you add an example using this property in sandbox to have all variants there?

**Comment by saledjenic** [comment]
@noeliaSD thanks for the comments, do you think that `leftSide` is more appropriate and more descriptive than `leftSideCheckbox`?

As of space before `?` in a ternary operator, personally I am always more for the compact mode, without a space before `?`, so is it mandatory to have a space in such cases in statusq?

**Comment by saledjenic** [comment]
@noeliaSD prop renamed to `leftSide`, an example using `StatusCheckbox` added to sandbox app.

**Review by caybro — APPROVED** [review]
This will be fun to handle with RTL languages :D

**Comment by noeliaSD** [comment]
> @noeliaSD prop renamed to `leftSide`, an example using `StatusCheckbox` added to sandbox app.

Thanks!! :)

**Pull request merged**